src/pages/evm/nfts/NftInventoryPage.vue

```diff
@@ -101,16 +108,69 @@ const tableRows = computed(() => {
         collectionAddress: nft.contractAddress,
     }));
 });
-const listImagesLoadingStates = ref<Record<string, boolean>>({});
+const showNoFilteredResultsState = computed(() => (collectionFilter.value || searchFilter.value) && !nftsToShow.value.length);
 
 
 // watchers
+watch(nftsAndCollectionListLoaded, (loaded) => {
+    if (loaded) {
+        nfts.value.forEach((nft) => {
+            // if an NFT icon in list view hasn't yet been loaded, enable the loading state for that image
+            // the loading state will be ended when the @loaded event is fired by that image
+            if (!showNftsAsTiles.value && listImagesLoadingStates.value?.[nft.id] !== false) {
+                listImagesLoadingStates.value[nft.id] = true;
+            }
+        });
+
+        // if this is initial load and there are pagination query params, validate and apply them
+        const { rowsPerPage, page, collection, search } = route.query;
+
+        let newRowsPerPage = 12;
+        let newPage = 1;
+
+        if (rowsPerPage && rowsPerPageOptions.includes(+(rowsPerPage ?? 12))) {
+            newRowsPerPage = +(rowsPerPage ?? 12);
+        }
+
+        // if there are no collection or search query params, and the page query param is valid (not greater than the total number of pages), apply it
+        if (!(collection || search) && page && (newRowsPerPage * (+(page ?? 1) - 1)) < nfts.value.length) {
+            newPage = +(page ?? 1);
+        }
```

Comment:
Is there any important reason why you prevent taking the page param when the filters ar "dirty"?
Because if not, I find it very useful to have everything taken from the url, specially when you press back button from detail page.

Example:
- goto: http://localhost:8081/evm/collectible-inventory?rowsPerPage=6&collection=0x386b18e01452884340DB7e5b6b0281f67b0B2C4b&search=12
- got to the next page (page 2)
- click on any NFT to see details
- Press the back button
  - my expected behavior would be to return to page 2 but it redirects me to page 1 because the filters are dirty.

If there's no reason to do this, I would suggest removing that condition like this:

```suggestion
        if (page && (newRowsPerPage * (+(page ?? 1) - 1)) < nfts.value.length) {
            newPage = +(page ?? 1);
        }
```

Reply:
yeah the issue is that, if i dont do this, then if you do the following steps you get an incorrect 'no results found' state.
1. clear all filters
2. go to page 2
3. add a filter, like collection = Ice Cream Zombies
4. no results found (because you are on page 2 but there is only 1 page of results)

maybe i can modify the logic so that the page persists only if there is more than 1 page of data. i will think about it, thank you

Reply:
nevermind i think that issue does not exist after some refactoring i did. good suggestion